spring-integration-gcp/src/test/java/org/springframework/integration/gcp/pubsub/outbound/PubSubMessageHandlerTests.java

```diff
@@ -69,4 +74,15 @@ public void testPublish() {
 						eq(ByteString.copyFrom("testPayload", Charset.defaultCharset())),
 						isA(Map.class));
 	}
+
+	@Test
+	public void testPublishSync() {
+		this.adapter.setSync(true);
+		Expression timeout = spy(this.adapter.getPublishTimeoutExpression());
+		this.adapter.setPublishTimeoutExpression(timeout);
+
+		this.adapter.handleMessage(this.message);
+		verify(timeout, times(1)).getValue(
+				eq(null), eq(this.message), eq(Long.class));
```

Comment:
@artembilan I couldn't find a way to mock the `onInit()` behaviour, so I just put `null` as the first arg for `evaluationContext`. Let me know if there is anything I could do here!

Reply:
Well, I would mock `PubsubTemplate` and verify that it is invoked with the proper args and return value .
But this is fine since we ensure that expression is called downstream.

Reply:
Yeah. The previous test ensures the `PubSubTemplate` call is done. I didn't think it'd be necessary here since that happens outside of the `if (this.sync)` block.